### PR TITLE
Improve auto-updater behavior on Windows (WEBAPP-3544)

### PR DESCRIPTION
**Pull request by gregor**
Related to https://github.com/wireapp/wire-desktop/pull/336

- will listen to updated versions of the Wire for Windows version
- adds a current version reporting on the webapp server side
- browser will check against that version every 6 hours
- bar informing the user will be shown at that time
- click reloads the webapp or restarts the Windows wrapper

### CODE REVIEW

File: app/script/event/WebApp.coffee [hunk 1]
```diff
@@ -109,8 +108,16 @@ z.event.WebApp =
   LEFT:
     HIDE: 'wire.webapp.left.hide'
     FADE_IN: 'wire.webapp.left.fade-in'
+  LIFECYCLE:
+    ASK_TO_CLEAR_DATA: 'wire.webapp.lifecycle.ask-to-clear-data'
+    LOADED: 'wire.webapp.lifecycle.loaded'
+    REFRESH: 'wire.webapp.lifecycle.refresh'
+    RESTART: 'wire.webapp.lifecycle.restart'
+    SIGN_OUT: 'wire.webapp.lifecycle.sign_out'
```
Comment (bennycode) [inline app/script/event/WebApp.coffee:116]: All the other values here are with `-`, so `sign_out` should also be `sign-out`.

Comment (gregor) [reply app/script/event/WebApp.coffee:116]: Unfortunately not. It's a mix of both in the file. `sign_out` has the upside of marking the full string part on double-click.

File: app/script/event/WebApp.coffee [hunk 2]
```diff
@@ -109,8 +108,16 @@ z.event.WebApp =
...
+    UPDATE: 'wire.webapp.lifecycle.update'
+  LOADED: 'wire.webapp.loaded' # todo: deprecated - remove when user base of wrappers version >= 2.12 is large enough
   LOGOUT:
-    ASK_TO_CLEAR_DATA: 'wire.webapp.logout.ask-to-clear-data'
+    ASK_TO_CLEAR_DATA: 'wire.webapp.logout.ask-to-clear-data' # todo: deprecated - remove when user base of wrappers version >= 2.12 is large enough
   PREFERENCES:
```
Comment (herrmannplatz) [inline app/script/event/WebApp.coffee:121]: was just about to ask :) 👍

File: app/script/main/app.coffee [hunk 3]
```diff
@@ -437,6 +441,15 @@ class z.main.App
       @logger.warn 'No internet access. Continuing when internet connectivity regained.'
       $(window).on 'online', -> _logout_on_backend()
 
+  refresh: ->
+    if z.util.Environment.electron and z.util.Environment.os.win
+      amplify.publish z.event.WebApp.LIFECYCLE.RESTART
```
Comment (gregor) [inline app/script/main/app.coffee:445]: No return here on purpose as none suppporting wrappers will just reload instead.

File: app/style/warnings.less [hunk 4]
```diff
@@ -91,6 +95,11 @@
   margin-right: 8px;
 }
 
+.warning-bar-icon-small {
```
Comment (gregor) [inline app/style/warnings.less]: Obsolete, will remove

File: grunt/tasks/aws.coffee [hunk 5]
```diff
@@ -42,4 +42,4 @@ module.exports = (grunt) ->
     grunt.file.write path.join('aws', 'version'), grunt.option 'version'
 
   grunt.registerTask 'aws_prepare',      ['aws_version_file', 'clean:aws', 'copy:aws', 'copy:aws_templates', 'clean:aws_app', 'clean:aws_s3', 'compress:aws']
-  grunt.registerTask 'aws_run',          ['init', 'prepare_prod', 'aws_prepare', 'open:aws', 'shell:aws']
+  grunt.registerTask 'aws_run',          ['init', 'prepare_prod', 'aws_prepare', 'open:aws', 'shell:aws', 'watch']
```
Comment (gregor) [inline grunt/tasks/aws.coffee]: Does not work as expected due to CSPs included in sever setup, will revert

File: app/script/main/app.coffee [hunk 6]
```diff
@@ -212,7 +214,9 @@ class z.main.App
       @_show_ui()
       @telemetry.time_step z.telemetry.app_init.AppInitTimingsStep.SHOWING_UI
       @telemetry.report()
+      # todo: deprecated - remove after update of supporting wrappers
```
Comment (gregor) [inline app/script/main/app.coffee]: Improve depraction message to include specific wrapper version

File: app/script/announce/AnnounceRepository.coffee [hunk 7]
```diff
@@ -30,20 +32,26 @@ class z.announce.AnnounceRepository
...
+  check_announcements: =>
     @announce_service.get_announcements()
     .then @process_announce_list
     .catch (error) =>
       @logger.error "Failed to fetch announcements: #{error}"
 
+  check_version: =>
+    @announce_service.get_version()
+    .then (server_version) ->
+      amplify.publish z.event.WebApp.LIFECYCLE.UPDATE if server_version > z.util.Environment.version false, true
```
Comment (lipis) [inline app/script/announce/AnnounceRepository.coffee]: This is confusing I have to admit.. Maybe something like 
```
needs_update = server_version > z.util.Environment.version...
```

and then user the variable?

**Pull request merged**